### PR TITLE
[SW-2791] Add Spark 3.4 support

### LINKED ISSUE #5618: Support for Spark 3.4
Hi,

any plans to support Spark 3.4?

Release notes: https://spark.apache.org/releases/spark-release-3-4-0.html

Cheers

### PR DESCRIPTION
**Pull request by krasinski**
closes #5618

### CODE REVIEW

**Comment by h2o-ops** [comment]
Hi @krasinski, your PR title "Add Spark 3.4 support" is missing JIRA issue number! Please specify it in the following form [SW-]

**Comment by h2o-ops** [comment]
Hi @krasinski, your PR title "Add Spark 3.4 support" is missing JIRA issue number! Please specify it in the following form [SW-]

**Comment by h2o-ops** [comment]
Hi @krasinski, your PR title "Add Spark 3.4 support" is missing JIRA issue number! Please specify it in the following form [SW-]

**Comment by h2o-ops** [comment]
Hi @krasinski, your PR title "Add Spark 3.4 support" is missing JIRA issue number! Please specify it in the following form [SW-]

File: ml/src/test/scala/ai/h2o/sparkling/ml/models/H2OMOJOModelTestSuite.scala [hunk 1]
```diff
@@ -110,19 +110,21 @@ class H2OMOJOModelTestSuite extends FunSuite with SharedH2OTestContext with Matc
       TestUtils.assertDataFramesAreIdentical(expectedCVScoringHistory(i), deserializedCVScoringHistory(i))
     }
   }
+  //TODO temporary Ignore - failing on Spark 3.4 - StructType incompatible class change
+  if (!createSparkSession().version.startsWith("3.4")) {
+    ignore("H2OMOJOModel saved with scala 2.11 behaves the same way as H2OMOJOModel saved with scala 2.12") {
```
Comment (mn-mikke) [inline ml/src/test/scala/ai/h2o/sparkling/ml/models/H2OMOJOModelTestSuite.scala:115]: Should this be `test` instead of `ignore`?

**Review by mn-mikke — APPROVED** [review]
LGTM

**Pull request merged**